typing/typecore.ml

```diff
@@ -1418,7 +1418,7 @@ and type_pat_aux ~constrs ~labels ~no_existentials ~mode ~explode ~env
       let id, _tme, new_env =
         !type_open Asttypes.Fresh !env sp.ppat_loc me in begin
       match id with
-      | Some _ -> assert false
+      | Some _ -> assert false (* not possible due to syntactic restriction *)
```

Comment:
We are trying to avoid in the type-checker implicit assumptions about the AST enforced by the parser (and that could be broken by alternative syntaxes or ppx).  Can you extend Ast_invariant with the corresponding check?

Reply:
Yes, with the explicit check in Ast_invariant  and the comment, this is fine.

Reply:
done